NoQuD/HomogeneousSolve/NodalSolve.py

```diff
@@ -0,0 +1,254 @@
+
+import numpy as np
+import matplotlib.pyplot as plt
+from NoQuD.homogenize.homogenize import *
+from NoQuD.nodal.Nodal import *
+from numpy.polynomial.legendre import legval as legval
+
+
+class NodalSolve:
+
+    """ Creates homogenized data for a 1D, 2 group problem and solves using nodal quasi-diffusion methods.
+
+    The homogenized nuclear data for the assembly(s) described by the input files is calculated. From this homogenized
+    data, the linear system required for a nodal solve is generated. Variables necessary for the solve process are
+    initialized.
+
+    Args:
+        assembly_input_files (str[]):  should contain the assembly information file, followed by the assembly information
+        files for each unique assembly.
+
+    Attributes:
+        homogenized_problem (HomogenizeGlobe instance): contains homogenized nuclear data.
+        assembly_sizes (float[][]): size of each unique assembly in each group
+        nodal_data (Nodal instance): contains the linear system necessary for the nodal solve method
+        order (int): number of coefficients in polynomial approximation to flux in each node
+        groups (int): number of energy groups in problem (current implementation only works for two groups)
+        nodes (int): number of nodes in problem. Equal to number of assemblies present in heterogeneous problem.
+        solution_cells (int): mesh length in homogeneous solution. This is a per node quantity.
+        fast_system (float[][]): the linear system used to solve for the fast flux.
+        slow_system (float[][]): the linear system used to solve for the slow, or thermal, flux.
+        fission_source (float[][]): the fission source terms in the RHS of the nodal solve
+        scatter_source (float[][]): the scatter source terms in the RHS of the nodal solve.
+        k (float[]): the multiplication factor. Stores the current value and two previous values.
+        flux_coefficients (float[][]): coefficients to the polynomial flux approximation
+        flux (float[][][]): the fast and thermal fluxes across the nodes. Stores and current and previous values.
+        spatial_fission_source (float[][][]): the fission source across the nodes. Stores the current and previous
+            values.
+        chi (float[]): the probability for a fission neutron to be born into a particular energy group.
+        nu (float[]): the average number of neutrons produced per fission in each energy group.
+        flux_epsilon (float): a convergence parameter on the flux
+        k_epsilon (float): a convergence parameter on the mulitplication factor.
+        converged (boolean): evaluates whether all convergence parameters are met.
+        k_converged (boolean): evaluation of multiplication factor convergence.
+        flux_converged (boolean): evaluation of flux convergence.
+        fission_source_converged (boolean): evaluation of fission source convergence.
+    """
+
```

Comment:
This is a really large class. It might be better to break it up into multiple classes that are a bit more direct.